samples/vboxwrapper/vbox_common.cpp

```diff
@@ -262,51 +283,206 @@ void VBOX_BASE::dump_hypervisor_logs(bool include_error_logs) {
+		    hardware acceleration for virtual machines, but your processor does not support the required feature.\n \
+		    Please report this issue to the project so that it can be addresssed.\n \
+		    Error Code: ERR_CPU_VM_EXTENSIONS_DISABLED\n",
+		
+		"   VboxSvc crashed while attempting to restore the current snapshot.  This is a critical\n \
+		    operation and this job cannot be recovered.\n",
+		
+		"   NOTE: VM session lock error encountered.\n \
+		    BOINC will be notified that it needs to clean up the environment.\n \
+		    This might be a temporary problem and so this job will be rescheduled for another time.\n",
+		
+		"   NOTE: BOINC has detected that your computer's processor supports hardware acceleration for\n \
+		    virtual machines but the hypervisor failed to successfully launch with this feature enabled.\n \
+		    This means that the hardware acceleration feature has been disabled in the computer's BIOS.\n \
+		    Please enable this feature in your computer's BIOS.\n \
+		    Intel calls it 'VT-x'\n \
+		    AMD calls it 'AMD-V'\n \
+		    More information can be found here: https://en.wikipedia.org/wiki/X86_virtualization\n \
+		    Error Code: ERR_CPU_VM_EXTENSIONS_DISABLED\n",
+		
+		"   NOTE: VirtualBox has reported an improperly configured virtual machine. It was configured to require\n \
+		    hardware acceleration for virtual machines, but your processor does not support the required feature.\n \
+		    Please report this issue to the project so that it can be addresssed.\n",
+		
+		"   NOTE: VirtualBox hypervisor reports that another hypervisor has locked the hardware acceleration\n \
+		    for virtual machines feature in exclusive mode.\n",
+		
+		"   NOTE: VirtualBox has failed to allocate enough memory to start the configured virtual machine.\n \
+		    This might be a temporary problem and so this job will be rescheduled for another time.\n",
+		
+		"   NOTE: VM failed to enter an online state within the timeout period.\n \
+		    This might be a temporary problem and so this job will be rescheduled for another time.\n",
+	
+		    "VM environment needed to be cleaned up.",
+       
+		    "Forign VM Hypervisor locked hardware acceleration features.",
+
+		    "VM Hypervisor was unable to allocate enough memory to start VM.",
+
+	     	"VM Hypervisor failed to enter an online state in a timely fashion."
+	};
+
+	std::vector<std::string> v(args, args + 14);
+	
+	return v[num];
+}
+
+string VBOX_BASE::read_vm_log(){
+	string line;
+	size_t line_pos;
+	size_t line_end;
+	size_t line_start;
+	string msg;
+	string virtualbox_vm_log;
+	virtualbox_vm_log = vm_master_name + "/Logs/VBox.log";
+	
+
+	std::ifstream  src(virtualbox_vm_log.c_str(), std::ios::binary);
+	    
+	if ((src.is_open()) && (src.seekg(log_pointer))) {
+
+            while (std::getline(src, line)) {
+
+                line_start = line.find("Console:");
```

Comment:
In the one VBox.log that I have at hand I have this line in it:

    97:40:20.193248 Console::powerDown(): A request to power off the VM has been issued (mMachineState=Saving, InUninit=0)

How does the rest of the vboxwrapper behave if this happens to be the last line with `Console:` in it and `state` then becomes garbage?